src/pdf.rs

```diff
@@ -48,6 +69,36 @@ impl File {
             ffi::cairo_pdf_surface_set_size(self.inner.to_raw_none(), width, height);
         }
     }
+
+    pub fn set_metadata(&self, metadata: PdfMetadata, value: &str) {
+        unsafe {
+            ffi::cairo_pdf_surface_set_metadata(self.inner.to_raw_none(), metadata.into(), value.as_ptr() as _);
```

Comment:
This does not work (in general). `&str` is not `\0`-terminated so you can't just use `as_ptr()` on it like this.

There's `CStr` / `CString` for working with this correctly.

Reply:
Please check the whole cairo codebase for this while you're at it, and also especially the code of your previews PR :)

Reply:
Ok